migration-engine/migration-engine-tests/tests/create_migration/create_migration_tests.rs

```diff
@@ -321,6 +321,171 @@ fn creating_a_migration_with_a_non_existent_migrations_directory_should_work(api
+                    DROP TYPE "Mood_old";
+                    ALTER TABLE "Cat" ALTER COLUMN "mood" SET DEFAULT 'MOODY';
+                    COMMIT;
+
+                    -- AlterTable
+                    ALTER TABLE "Cat" ALTER COLUMN "mood" SET DEFAULT 'MOODY';
+                "#]]
+            } else {
+                panic!()
+            };
+
+            migration.expect_contents(expected_script)
+        });
+}
+
+#[test_connector(tags(Postgres, CockroachDb))]
+fn alter_enum_name_with_default_empty_list_shouldnt_set_new_default(mut api: TestApi) {
+    let plain_dm = api.datamodel_with_provider(
+        r#"
+        model Cat {
+            id      Int    @id
+            moods   Mood[] @default([])
+        }
+
+        enum Mood {
+            HUNGRY
+            SLEEPY
+        }
+    "#,
+    );
+
+    let dir = api.create_migrations_directory();
+    api.create_migration("plain", &plain_dm, &dir).send_sync();
+
+    let custom_dm = api.datamodel_with_provider(
+        r#"
+        model Cat {
+            id      Int    @id
+            moods   Mood[] @default([])
+        }
+
+        enum Mood {
+            SLEEPY
+            MOODY
+        }
+    "#,
+    );
+
+    let is_postgres = api.is_postgres();
+    let is_cockroach = api.is_cockroach();
+
+    api.create_migration("custom", &custom_dm, &dir)
+        .send_sync()
+        .assert_migration_directories_count(2)
+        .assert_migration("custom", move |migration| {
+            let expected_script = if is_cockroach {
+                expect![[r#"
+                    /*
+                      Warnings:
+
+                      - The values [HUNGRY] on the enum `Mood` will be removed. If these variants are still used in the database, this will fail.
+
+                    */
+                    -- AlterEnum
+                    ALTER TYPE "Mood" ADD VALUE 'MOODY';
+                    ALTER TYPE "Mood" DROP VALUE 'HUNGRY';
+                "#]]
+            } else if is_postgres {
+                expect![[r#"
+                    /*
+                      Warnings:
+
+                      - The values [HUNGRY] on the enum `Mood` will be removed. If these variants are still used in the database, this will fail.
+
+                    */
+                    -- AlterEnum
+                    BEGIN;
+                    CREATE TYPE "Mood_new" AS ENUM ('SLEEPY', 'MOODY');
+                    ALTER TABLE "Cat" ALTER COLUMN "moods" DROP DEFAULT;
+                    ALTER TABLE "Cat" ALTER COLUMN "moods" TYPE "Mood_new"[] USING ("moods"::text::"Mood_new"[]);
+                    ALTER TYPE "Mood" RENAME TO "Mood_old";
+                    ALTER TYPE "Mood_new" RENAME TO "Mood";
+                    DROP TYPE "Mood_old";
+                    ALTER TABLE "Cat" ALTER COLUMN "moods";
+                    COMMIT;
+                "#]]
+            } else {
+                panic!()
+            };
+
+            migration.expect_contents(expected_script)
+        });
+}
```

Comment:
Could we test the following scenario:

- Single model with a field with `            mood    Mood @default(HUNGRY)`
- Migration to drop one variant from the enum and the default on the field at the same time.

I think it's a more minimal reproduction of the conditions for the crash.

That test can live in tests/migrations/enums.rs

Reply:
Isn't this scenario already tested by `alter_enum_name_remove_and_change_default`?

Reply:
yep we could move it there. It's a different test however, `alter_enum_name_remove_and_change_default` still has a default, just a different one. We should test what happens when the whole `@default()` is removed in the second schema.

Reply:
I have moved the tests to `tests/migrations/enums.rs` and left a few TODOs.